load.go

```diff
@@ -447,3 +455,75 @@ func (b *ArtifactBuild) processBuildArgs(lex *shell.Lex, args map[string]string,
 
 	return goerrors.Join(errs...)
 }
+
+func (i *ImageConfig) validate() error {
+	if i == nil {
+		return nil
+	}
+
+	var errs []error
+	if err := i.Post.validate(); err != nil {
+		errs = append(errs, errors.Wrap(err, "postinstall"))
+	}
+
+	return goerrors.Join(errs...)
+}
+
+func (p *PostInstall) validate() error {
+	if p == nil {
+		return nil
+	}
+
+	var errs []error
+
+	if err := validateSymlinks(p.Symlinks); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.Wrap(goerrors.Join(errs...), "symlink")
+}
+
+func validateSymlinks(symlinks map[string]SymlinkTarget) error {
+	var errs []error
+	dests := make(map[string]string, len(symlinks)<<1)
+
+	for oldpath, cfg := range symlinks {
+		var err error
+		if oldpath == "" {
+			err = fmt.Errorf("symlink source is empty")
+			errs = append(errs, err)
+		}
+
+		if cfg.Path != "" && len(cfg.Paths) != 0 || cfg.Path == "" && len(cfg.Paths) == 0 {
+			err = fmt.Errorf("'path' and 'paths' fields are mutually exclusive, and at least one is required")
+			errs = append(errs, err)
+		}
+
+		if err != nil {
+			continue
+		}
+
+		// By this point, both the oldpath and the SymlinkTarget are
+		// well-formed. We still need to make sure each 'newpath' is unique.
+		if cfg.Paths == nil {
+			cfg.Paths = []string{}
+		}
+
+		newpaths := slices.Clone(cfg.Paths)
```

Comment:
Why the clone? Should not need to modify anything here.
If we want to make sure that `cfg.Path` is not also somewhere in `cfg.Paths` you can check that directly.

```go
for _, link := range symlinks {
  if link.Path == cfg.Path {
    // error
  }
}
```

Reply:
I could have sworn I put this in `fillDefaults`. I need to move this code there. The clone from moving the old code around. I'll take another pass

Reply:
> Why the clone? Should not need to modify anything here. If we want to make sure that `cfg.Path` is not also somewhere in `cfg.Paths` you can check that directly.
> 
> ```go
> for _, link := range symlinks {
>   if link.Path == cfg.Path {
>     // error
>   }
> }
> ```

And the goal is to make sure the there are no duplicate `oldpath`s in **the whole `map[string]SymlinkTarget`**. The desired behavior is unclear (follow the symlink that has already been created? Or overwrite the existing symlink?) Map iteration is not reliably ordered (and we sort the keys to avoid cache misses), so the user is unlikely to get what they want anyway. Better to just not allow it.

Reply:
done. This exhibits the desired behavior, doesn't modify the spec, is tested properly, and only allocates the map when needed.